Source/Core/Common/Logging/LogManager.cpp

```diff
@@ -100,18 +100,33 @@ LogManager::LogManager()
   IniFile::Section* options = ini.GetOrCreateSection("Options");
   bool write_file;
   bool write_console;
+  bool write_window;
   options->Get("WriteToFile", &write_file, false);
   options->Get("WriteToConsole", &write_console, true);
+  options->Get("WriteToWindow", &write_window, true);
+
+  // Set up log listeners
+  int verbosity;
+  options->Get("Verbosity", &verbosity, 0);
+
+  // Ensure the verbosity level is valid
+  if (verbosity < 1)
+    verbosity = 1;
+  if (verbosity > MAX_LOGLEVEL)
+    verbosity = MAX_LOGLEVEL;
 
   for (LogContainer* container : m_Log)
   {
     bool enable;
     logs->Get(container->GetShortName(), &enable, false);
     container->SetEnable(enable);
+    container->SetLevel(static_cast<LogTypes::LOG_LEVELS>(verbosity));
     if (enable && write_file)
       container->AddListener(LogListener::FILE_LISTENER);
     if (enable && write_console)
       container->AddListener(LogListener::CONSOLE_LISTENER);
+    if (enable && write_window)
+      container->AddListener(LogListener::LOG_WINDOW_LISTENER);
```

Comment:
...crash. Whoops, looks like I missed that no window listener would be registered with NoGUI.

I have added a check to make sure listeners exist before using them.